### ISSUE TITLE
Stack navigator's transitionStart doesn't trigger on Android

### ISSUE THREAD

**Issue by AndreasWintherMoen** [react-navigation/react-navigation#11821]
### Current behavior

The navigation event _transitionStart_ triggers on iOS but on Android. See code below and [minimal Snack](https://snack.expo.dev/@andw/courageous-yellow-cookies).

```
  const navigation = useNavigation();
  
  useEffect(() => {
    navigation.addListener("transitionStart", onBackPress); // the function runs on iOS but not Android
    return () => {
      navigation.removeListener("transitionStart", onBackPress);
    };
  }, [navigation, onBackPress]);
```

### Expected behavior

_transitionStart_ navigation event should trigger on all platforms. 

### Reproduction

https://snack.expo.dev/@andw/courageous-yellow-cookies

### Platform

- [X] Android
- [ ] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-top-tabs
- [ ] @react-navigation/stack
- [X] @react-navigation/native-stack
- [ ] react-native-tab-view

### Environment

- [x] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               | ^6.1.9
| @react-navigation/native-stack         | ^6.9.17
| react-native-safe-area-context         | 4.6.3
| react-native-screens                   | ~3.22.0
| react-native                           | 0.72.6
| expo                                   | ~49.0.15
| node                                   | v20.5.1
| npm                             | 9.8.0

**Comment by AndreasWintherMoen** [comment]
> The versions mentioned in the issue for the following packages differ from the latest versions on npm:
> 
> * `@react-navigation/native` (found: `6.1.9`, latest: `6.1.10`)
> 
> Can you verify that the issue still exists after upgrading to the latest versions of these packages?

Issue still exists after updating.

**Comment by mobinni** [comment]
We are seeing the same issue as well, have you found a fix?

**Comment by AndreasWintherMoen** [comment]
> We are seeing the same issue as well, have you found a fix?

No, unfortunately. I ended up switching to @react-navigation/stack instead.

**Comment by Gustash** [comment]
This issue still exists as of:

```
react-navigation/native@7.0.14
react-navigation/native-stack@7.2.0
```

The `transitionEnd` event also doesn't fire from the NativeStackNavigator.
